### PR TITLE
Pre-Heal Update

### PR DESCRIPTION
**Pull request by aumu1031**
- Pre-Heal to 95%
- Healing items are now a %
- Cap out of combat healing to pre-heal limit
- Added configuration for per-item carry limit
- Reworked router logic so we can configure conditional overrides for particular pages that are otherwise set to "only in village".

### CODE REVIEW

File: config/routes.php
Comment (levimeahan) [inline config/routes.php:138]: I think this is the point we should move away from `village_ok` entirely to a more granular `allowed_location_types` array that maps from location type => allowed or not

```suggestion
        allowed_location_types: [
            Route::LOCATION_VILLAGE,
            Route::LOCATION_COLOSSEUM
        ],
```
If this param is not provided, we could assume all location types are okay. But if provided, we only let the route trigger if the current location type matches one in this array

**Review by levimeahan — APPROVED** [review]
Approving to unblock, but please address the route location type feedback before merging 👍

File: config/routes.php
Comment (aumu1031) [reply config/routes.php:138]: I like `allowed_location-types` solution much better, building on `village_ok` logic makes this overly complicated.
I will rework this and apply to all existing routes in a follow up.

**Review by levimeahan — APPROVED** [review]
Awesome, this is much better!

**Pull request merged**
